llvm/lib/Target/RISCV/RISCVTargetTransformInfo.cpp

```diff
@@ -1436,6 +1436,13 @@ InstructionCost RISCVTTIImpl::getCmpSelInstrCost(unsigned Opcode, Type *ValTy,
     }
   }
 
+  // The cmp + select instructions will lower to SELECT_CC and lower to
+  // PseudoCCMOVGPR which will generate a conditional branch + mv. The
+  // cost of (cmp + select) will be (0 + select inst cost).
+  if (I && isa<CmpInst>(I) && I->hasOneUser() &&
```

Comment:
Should this be ICmpInst not CmpInst? CmpInst includes FCmpInst which isn't free.

I think you may also need to check that the ICmpInst type is scalar and the select type is scalar. A vector ICmp is not free. And a vector select does not lower PseudoCCMOVGPR.

Reply:
Fixed. Thanks!